api/types/resource.go

```diff
@@ -452,6 +495,56 @@ func (m *Metadata) SetOrigin(origin string) {
 	m.Labels[OriginLabel] = origin
 }
 
+// IsEqual determines if two metadata resources are equivalent to one another.
+func (m *Metadata) IsEqual(i interface{}) bool {
```

Comment:
What is the practical reason for having these functions support both pointer and non-pointer objects? I understand that in theory it may be convenient to not have to care what argument you give it but practically is there any scenario you would say, want to pass a non-pointer here?

If we got rid of `interface{}`, those methods would simplify quite a bit.

Reply:
We can't really avoid interface here because we can't do a type assertion of something like `IsEqual[T]` here, unfortunately.

I made the difference primarily because I don't know for sure if messages will always be pointers. I'm willing to alter this you feel strongly, however.